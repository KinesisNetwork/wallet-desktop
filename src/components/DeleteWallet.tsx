import * as React from 'react'

import { getPasswordConfirmation } from '@components/PasswordConfirmation'
import { Wallet } from '@types'

export interface Props {
  activeWallet: Wallet
  deleteWallet: (wallet: Wallet) => any
}

export class DeleteWallet extends React.Component<Props> {
  displayName = 'DeleteWallet'

  deleteWallet = async () => {
    const isSureToDelete = await sweetAlert({
      buttons: ['Go back', 'Delete Wallet'],
      dangerMode: true,
      icon: 'warning',
      text: 'Once deleted, you will not be able to recover this wallet.',
      title: 'Are you sure?',
    })
    if (!isSureToDelete) {
      return
    }
    const continueWithDelete = await getPasswordConfirmation(this.props.activeWallet)
    if (continueWithDelete) {
      this.props.deleteWallet(this.props.activeWallet)
    }
  }

  render() {
    return (
      <div className='buttons is-hidden-mobile'>
        <button type='submit' className='button is-danger' onClick={() => this.deleteWallet()}>
          <span className='icon'>
            <i className='fa fa-trash-alt fa-lg' />
          </span>
          <span>Delete Account from Wallet</span>
        </button>
      </div>
    )
  }
}
