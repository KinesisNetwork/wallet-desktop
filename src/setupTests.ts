import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
