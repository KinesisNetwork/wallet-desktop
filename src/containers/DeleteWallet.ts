import { connect } from 'react-redux'

import { deleteWallet } from '@actions'
import { DeleteWallet as DeleteWalletPresentation } from '@components/DeleteWallet'
import { RootState } from '@store'

const mapStateToProps = ({ wallets: { activeWallet } }: RootState) => ({
  activeWallet,
})

export const DeleteWallet = connect(
  mapStateToProps,
  { deleteWallet },
)(DeleteWalletPresentation)
