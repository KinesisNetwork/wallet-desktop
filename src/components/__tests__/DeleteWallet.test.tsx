import { shallow } from 'enzyme'
import * as React from 'react'

import { DeleteWallet } from '@components/DeleteWallet'
import '../../setupTests'

describe('DeleteWallet', () => {
  let props

  beforeEach(() => {
    props = {
      activeWallet: {
        accountName: 'hello',
        encryptedPrivateKey: 'uhgn',
        publicKey: 'qwe',
      },
      deleteWallet: () => null,
    }
  })
  it('should render correctly', () => {
    const wrapper = shallow(<DeleteWallet {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders a button', () => {
    const wrapper = shallow(<DeleteWallet {...props} />)

    expect(wrapper.find('button')).toHaveLength(1)
    expect(wrapper.find('button').text()).toEqual('Delete Account from Wallet')
    expect(
      wrapper
        .find('.icon')
        .children()
        .hasClass('fa-trash-alt'),
    ).toBe(true)
    expect(
      wrapper
        .find('.icon')
        .children()
        .hasClass('fa-lg'),
    ).toBe(true)
  })
})
