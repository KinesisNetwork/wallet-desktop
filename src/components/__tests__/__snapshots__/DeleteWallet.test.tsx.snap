// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteWallet should render correctly 1`] = `
<div
  className="buttons is-hidden-mobile"
>
  <button
    className="button is-danger"
    onClick={[Function]}
    type="submit"
  >
    <span
      className="icon"
    >
      <i
        className="fa fa-trash-alt fa-lg"
      />
    </span>
    <span>
      Delete Account from Wallet
    </span>
  </button>
</div>
`;
